io-engine/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -326,9 +326,10 @@ impl<'n> Nexus<'n> {
         // Close and remove the child.
         let res = match self.lookup_child(uri) {
             Some(child) => {
-                // Remove child from the I/O path.
+                // Detach the child from the I/O path, and close its handles.
                 if let Some(device) = child.get_device_name() {
-                    self.disconnect_device(&device).await;
+                    self.detach_device(&device).await;
+                    self.disconnect_all_detached_devices().await;
```

Comment:
Is there a race here, between pause and remove_child?

Reply:
IIRC we discussed with place: we are removing device which is not failed. Pause is done before disconnect, and will eventually succeed.